### PR TITLE
Validate the corretness of results when running matmul benchmarks

### PR DESCRIPTION
**Pull request by zasdfgbnm**

### CODE REVIEW

File: nvfuser/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+__init__.py
```
Comment (zasdfgbnm) [inline nvfuser/.gitignore:1]: This is not related, but I feel it is convenient because this is generated every time when I run tests...

Comment (naoyam) [reply nvfuser/.gitignore:1]: I have no idea what this file does, but it was removed recently (https://github.com/csarofeen/pytorch/pull/2337/files#diff-e512bea3b02f75ab1e81b759562879c5867e6e863679d6e7696fa34087dc3dc9L1). Is it still generated locally?

File: third_party/nvfuser/benchmark/matmul.cpp [hunk 2]
```diff
@@ -101,16 +101,21 @@ std::pair<at::Tensor, at::Tensor> fp16MatmulAtInput(
     MatmulLayout layout) {
   auto options = at::TensorOptions().dtype(at::kHalf).device(at::kCUDA, 0);
 
+  // Use randint to reduce numerical error so we can easily validate the
+  // correctness of results.
   switch (layout) {
     case MatmulLayout::TT:
       return std::make_pair(
-          at::randn({M, K}, options), at::randn({K, N}, options));
+          at::randint(-3, 3, {M, K}, options).div_(8),
```
Comment (naoyam) [inline third_party/nvfuser/benchmark/matmul.cpp:109]: Does this mean a 2D tensor of size `MxK` of values randomly generated between -3 and 3 and is then divided by 8? Why division by 8?

**Review by naoyam — COMMENTED** [review]
Was there any issue with the previous validation method?

File: third_party/nvfuser/benchmark/matmul.cpp (hunk 2 above)
Comment (zasdfgbnm) [reply third_party/nvfuser/benchmark/matmul.cpp:109]: Yes, you are right. I want values of 0.XX with small mantissa usage. Compared with multiplying `0.1`, 8 is a multiple of 2, so it uses less mantissa.

File: nvfuser/.gitignore (hunk 1 above)
Comment (zasdfgbnm) [reply nvfuser/.gitignore:1]: Yes, it keeps being regenerated.

**Comment by zasdfgbnm** [comment]
> Was there any issue with the previous validation method?

I think previously, we were not validating at all?

File: third_party/nvfuser/benchmark/matmul.cpp (hunk 2 above)
Comment (naoyam) [reply third_party/nvfuser/benchmark/matmul.cpp:109]: Doesn't `at::randint` generate an integer tensor, and division by 8 would just result in zero?

Comment (zasdfgbnm) [reply third_party/nvfuser/benchmark/matmul.cpp:109]: The dtype will be the dtype in `options`: See https://pytorch.org/docs/stable/generated/torch.randint.html

**Review by naoyam — APPROVED** [review]
LGTM

**Pull request merged**